assets/main.js.liquid

```diff
@@ -81,33 +81,189 @@ Site.Header = {
 };
 
 Site.Product = {
+  minWidth: 720, // minWidth for fixed header layout
+  numberRelatedToPick: 4, // number of related products to show
   init: function() {
     var _this = this;
 
-    if ($('#product-gallery').length) {
-      _this.setGalleryWidth();
+    _this.$container = $('.container'); // used by toggleFixed() on scroll
+
+    _this.$productHeader = $('#product-header');
+    _this.$productGallery = $('#product-gallery');
+
+    // imagesLoaded hasn't finished yet
+    _this.loadedImages = false;
+
+    if (_this.$productGallery.length) {
+      // We have product images, so we do the fancy layout
+      _this.handleLayout();
     }
 
     if ($('#related-products').length) {
+      // Pick a few Related Products to display
       _this.pickRelated();
     }
   },
 
-  setGalleryWidth: function() {
-    var galleryWidth = 0;
+  handleLayout: function() {
+    var _this = this;
 
-    $('.product-gallery-item').each(function() {
-      galleryWidth += $(this).width();
-    });
+    _this.windowWidth = $(window).width();
+
+    if (_this.windowWidth >= _this.minWidth) {
+      // window is wider than minimum width (720px) for fixed header layout
+
+      if (!_this.loadedImages) {
+        // bind imagesLoaded
+        _this.$productGallery.imagesLoaded( function() {
+          // remove min-height for product gallery holder
+          _this.$productGallery.removeClass('min-height');
+          // size image holder to fit image size
+          _this.sizeImageHolder();
+          // get dimensions for toggleFixed()
+          _this.getFixedHeaderDimensions();
+          // toggle fixed header
+          _this.toggleFixed();
```

Comment:
have you considered using callbacks for things like this `getFixedHeaderDimensions` and then `toggleFixed`. might make relation between functions clearer to read too

Reply:
I'm not sure how best to do that. you can implement if you want